lib/AnyEvent/Handle/UDP.pm

```diff
@@ -165,7 +165,7 @@ sub _bind_to {
 	my $bind_to = sub {
 		my ($domain, $type, $proto, $sockaddr) = @_;
 		if (!Scalar::Util::openhandle($fh)) {
-			socket $fh, $domain, $type, $proto or redo;
+			socket $fh, $domain, $type, $proto or die "Could not create socket: $!";
```

Comment:
Why are you mixing die() and _error() here? I know this is not exaktly the same, but i guess die() would now (thanks to eval) working properly (see also $connect_to).

Reply:
Fair point